File: wind_input/pkg/config/enums.go
// Package config: enums.go 定义配置中以字符串字面量形式存在的"行为模式"枚举类型。
//
// 设计原则：
//   - 使用 `type Foo string` + const 块，而非 iota 整数枚举，因这些值会进 YAML 配置。
//   - YAML/JSON tag 保持不变，序列化值与原字符串完全一致。
//   - 每种类型提供 `Valid() bool` 方法，便于配置加载/迁移时的校验。
package config

// EnterBehavior 回车键行为
type EnterBehavior string

const (
	EnterCommit         EnterBehavior = "commit"           // 上屏编码
	EnterClear          EnterBehavior = "clear"            // 清空编码
	EnterCommitAndInput EnterBehavior = "commit_and_input" // 上屏并继续输入
	EnterIgnore         EnterBehavior = "ignore"           // 忽略
)

// Valid 校验取值是否在合法集合内
func (b EnterBehavior) Valid() bool {
	switch b {
	case EnterCommit, EnterClear, EnterCommitAndInput, EnterIgnore:
		return true
	}
	return false
}

// SpaceOnEmptyBehavior 空码时空格键行为
type SpaceOnEmptyBehavior string

const (
	SpaceOnEmptyCommit         SpaceOnEmptyBehavior = "commit"
	SpaceOnEmptyClear          SpaceOnEmptyBehavior = "clear"
	SpaceOnEmptyCommitAndInput SpaceOnEmptyBehavior = "commit_and_input"
	SpaceOnEmptyIgnore         SpaceOnEmptyBehavior = "ignore"
)

// Valid 校验取值
func (b SpaceOnEmptyBehavior) Valid() bool {
	switch b {
	case SpaceOnEmptyCommit, SpaceOnEmptyClear, SpaceOnEmptyCommitAndInput, SpaceOnEmptyIgnore:
		return true
	}
	return false
}

// OverflowBehavior 候选按键无效时的处理策略（数字键/二三候选键/以词定字键共用）
type OverflowBehavior string

const (
	OverflowIgnore         OverflowBehavior = "ignore"
	OverflowCommit         OverflowBehavior = "commit"
	OverflowCommitAndInput OverflowBehavior = "commit_and_input"
)

// Valid 校验取值
func (b OverflowBehavior) Valid() bool {
	switch b {
	case OverflowIgnore, OverflowCommit, OverflowCommitAndInput:
		return true
	}
	return false
}

// FilterMode 候选过滤模式
type FilterMode string

const (
	FilterSmart   FilterMode = "smart"   // 智能（默认）
	FilterGeneral FilterMode = "general" // 仅常用字
	FilterGB18030 FilterMode = "gb18030" // 不限制
)

// Valid 校验取值
func (m FilterMode) Valid() bool {
	switch m {
	case FilterSmart, FilterGeneral, FilterGB18030:
		return true
	}
	return false
}

// ThemeStyle 主题风格
type ThemeStyle string

const (
	ThemeStyleSystem ThemeStyle = "system" // 跟随系统
	ThemeStyleLight  ThemeStyle = "light"  // 亮色
	ThemeStyleDark   ThemeStyle = "dark"   // 暗色
)

// Valid 校验取值
func (s ThemeStyle) Valid() bool {
	switch s {
	case ThemeStyleSystem, ThemeStyleLight, ThemeStyleDark:
		return true
	}
	return false
}

// CandidateLayout 候选布局
type CandidateLayout string

const (
	LayoutHorizontal CandidateLayout = "horizontal"
	LayoutVertical   CandidateLayout = "vertical"
)

// Valid 校验取值
func (l CandidateLayout) Valid() bool {
	switch l {
	case LayoutHorizontal, LayoutVertical:
		return true
	}
	return false
}

// PreeditMode 编码显示模式
type PreeditMode string

const (
	PreeditTop      PreeditMode = "top"      // 编码在上方独立行（默认）
	PreeditEmbedded PreeditMode = "embedded" // 嵌入候选行前
)

// Valid 校验取值
func (m PreeditMode) Valid() bool {
	switch m {
	case PreeditTop, PreeditEmbedded:
		return true
	}
	return false
}

// PinyinSeparatorMode 拼音分隔符模式
type PinyinSeparatorMode string

const (
	PinyinSeparatorAuto     PinyinSeparatorMode = "auto"
	PinyinSeparatorQuote    PinyinSeparatorMode = "quote"
	PinyinSeparatorBacktick PinyinSeparatorMode = "backtick"
	PinyinSeparatorNone     PinyinSeparatorMode = "none"
)

// Valid 校验取值
func (m PinyinSeparatorMode) Valid() bool {
	switch m {
	case PinyinSeparatorAuto, PinyinSeparatorQuote, PinyinSeparatorBacktick, PinyinSeparatorNone:
		return true
	}
	return false
}

// FontEngine 文本渲染引擎（对应 UIConfig.TextRenderMode 字段）
type FontEngine string

const (
	FontEngineDirectWrite FontEngine = "directwrite" // DirectWrite + Direct2D（默认）
	FontEngineGDI         FontEngine = "gdi"         // Windows GDI 原生
	FontEngineFreetype    FontEngine = "freetype"    // FreeType
)

// Valid 校验取值
func (e FontEngine) Valid() bool {
	switch e {
	case FontEngineDirectWrite, FontEngineGDI, FontEngineFreetype:
		return true
	}
	return false
}

// S2TVariant 简入繁出（简体→繁体）变体。
//
// 对应 OpenCC 配置链：
//   - s2t   ：标准繁体（不做地区词替换）
//   - s2tw  ：台湾正体（字形替换）
//   - s2twp ：台湾正体 + 词汇替换（如「软件」→「軟體」）
//   - s2hk  ：香港繁体（字形替换）
type S2TVariant string

const (
	S2TStandard     S2TVariant = "s2t"
	S2TTaiwan       S2TVariant = "s2tw"
	S2TTaiwanPhrase S2TVariant = "s2twp"
	S2THongKong     S2TVariant = "s2hk"
)

// Valid 校验取值
func (v S2TVariant) Valid() bool {
	switch v {
	case S2TStandard, S2TTaiwan, S2TTaiwanPhrase, S2THongKong:
		return true
	}
	return false
}

// PagerBarDisplay 翻页栏显示方式（用户级覆盖，优先级高于主题配置）
// 空字符串（PagerBarDefault）表示跟随主题配置。
type PagerBarDisplay string

const (
	PagerBarDefault PagerBarDisplay = ""       // 跟随主题
	PagerBarAlways  PagerBarDisplay = "always" // 总是显示翻页栏（含单页）
	PagerBarAuto    PagerBarDisplay = "auto"   // 大于一页时显示翻页栏
	PagerBarHide    PagerBarDisplay = "hide"   // 完全隐藏翻页栏（含箭头）
)

// Valid 校验取值
func (m PagerBarDisplay) Valid() bool {
	switch m {
	case PagerBarDefault, PagerBarAlways, PagerBarAuto, PagerBarHide:
		return true
	}
	return false
}

// PageNumberDisplay 页码文字显示方式（用户级覆盖，优先级高于主题配置）
// 空字符串（PageNumberDefault）表示跟随主题配置。
type PageNumberDisplay string

const (
	PageNumberDefault PageNumberDisplay = ""     // 跟随主题
	PageNumberShow    PageNumberDisplay = "show" // 显示页码文字
	PageNumberHide    PageNumberDisplay = "hide" // 隐藏页码文字
)

// Valid 校验取值
func (m PageNumberDisplay) Valid() bool {
	switch m {
	case PageNumberDefault, PageNumberShow, PageNumberHide:
		return true
	}
	return false
}

File: wind_setting/frontend/src/lib/enums.ts
// enums.ts — 前端配置枚举常量的单一权威来源。
//
// 这些常量值与后端 Go 端的字面量必须保持一致（见 wind_input/pkg/config/enums.go、
// wind_input/pkg/keys/keys.go、wind_input/pkg/keys/pair.go），因为它们是 YAML/JSON
// 协议字段值。本文件的目的是消除前端 Vue/TS 代码里散落的字符串字面量，提供集中
// 修改点。
//
// 用法：
//   import { EnterBehavior, FilterMode, ... } from '@/lib/enums';
//   if (cfg.enter_behavior === EnterBehavior.Commit) { ... }
//
// 注意：模板中的 `<SelectItem value="commit">` 等字面量出于"展示+本地化文案"
// 性质保留，但 `<script>` 中的逻辑比较应使用本文件的常量。

// ============================================================
// 行为枚举（对应 pkg/config/enums.go）
// ============================================================

export const EnterBehavior = {
  Commit: "commit",
  Clear: "clear",
  CommitAndInput: "commit_and_input",
  Ignore: "ignore",
} as const;
export type EnterBehaviorValue =
  (typeof EnterBehavior)[keyof typeof EnterBehavior];

export const SpaceOnEmptyBehavior = {
  Commit: "commit",
  Clear: "clear",
  CommitAndInput: "commit_and_input",
  Ignore: "ignore",
} as const;
export type SpaceOnEmptyBehaviorValue =
  (typeof SpaceOnEmptyBehavior)[keyof typeof SpaceOnEmptyBehavior];

export const OverflowBehavior = {
  Ignore: "ignore",
  Commit: "commit",
  CommitAndInput: "commit_and_input",
} as const;
export type OverflowBehaviorValue =
  (typeof OverflowBehavior)[keyof typeof OverflowBehavior];

export const FilterMode = {
  Smart: "smart",
  General: "general",
  GB18030: "gb18030",
} as const;
export type FilterModeValue = (typeof FilterMode)[keyof typeof FilterMode];

export const ThemeStyle = {
  System: "system",
  Light: "light",
  Dark: "dark",
} as const;
export type ThemeStyleValue = (typeof ThemeStyle)[keyof typeof ThemeStyle];

export const CandidateLayout = {
  Horizontal: "horizontal",
  Vertical: "vertical",
} as const;
export type CandidateLayoutValue =
  (typeof CandidateLayout)[keyof typeof CandidateLayout];

export const PreeditMode = {
  Top: "top",
  Embedded: "embedded",
} as const;
export type PreeditModeValue = (typeof PreeditMode)[keyof typeof PreeditMode];

export const PinyinSeparatorMode = {
  Auto: "auto",
  Quote: "quote",
  Backtick: "backtick",
  None: "none",
} as const;
export type PinyinSeparatorModeValue =
  (typeof PinyinSeparatorMode)[keyof typeof PinyinSeparatorMode];

export const FontEngine = {
  DirectWrite: "directwrite",
  GDI: "gdi",
  Freetype: "freetype",
} as const;
export type FontEngineValue = (typeof FontEngine)[keyof typeof FontEngine];

export const S2TVariant = {
  Standard: "s2t",       // 标准繁体
  Taiwan: "s2tw",        // 台湾繁体
  TaiwanPhrase: "s2twp", // 台湾繁体（含词汇）
  HongKong: "s2hk",      // 香港繁体
} as const;
export type S2TVariantValue = (typeof S2TVariant)[keyof typeof S2TVariant];

export const PagerBarDisplay = {
  Default: "",       // 跟随主题配置
  Hide: "hide",      // 完全隐藏翻页栏（含箭头）
  Auto: "auto",      // 大于一页时显示
  Always: "always",  // 总是显示翻页栏
} as const;
export type PagerBarDisplayValue =
  (typeof PagerBarDisplay)[keyof typeof PagerBarDisplay];

export const PageNumberDisplay = {
  Default: "",      // 跟随主题配置
  Show: "show",     // 显示页码文字
  Hide: "hide",     // 隐藏页码文字
} as const;
export type PageNumberDisplayValue =
  (typeof PageNumberDisplay)[keyof typeof PageNumberDisplay];

// ============================================================
// 修饰键 / 按键名（对应 pkg/keys/keys.go）
// ============================================================

export const Modifier = {
  Ctrl: "ctrl",
  Shift: "shift",
  Alt: "alt",
  Win: "win",
} as const;
export type ModifierValue = (typeof Modifier)[keyof typeof Modifier];

// 仅收录前端实际使用到的按键 token；如需扩充按 keys.go 同步添加。
// 值必须是 Go pkg/keys 的规范名（canonical），不可用别名——一致性由
// keysEnums.test.ts 对照 generated/keys.json 守卫（防止 "open_bracket" 之类别名漂移）。
export const Key = {
  // 字母（仅前端用到的）
  Z: "z",
  // 标点
  Semicolon: "semicolon",
  Quote: "quote",
  Comma: "comma",
  Period: "period",
  Slash: "slash",
  Backslash: "backslash",
  LBracket: "lbracket",
  RBracket: "rbracket",
  Grave: "grave",
  // 控制键
  Tab: "tab",
  // 修饰键作为独立 token
  LShift: "lshift",
  RShift: "rshift",
  LCtrl: "lctrl",
  RCtrl: "rctrl",
  CapsLock: "capslock",
} as const;
export type KeyValue = (typeof Key)[keyof typeof Key];

// ============================================================
// 组合键群（对应 pkg/keys/pair.go）
// ============================================================

export const PairGroup = {
  SemicolonQuote: "semicolon_quote",
  CommaPeriod: "comma_period",
  LRShift: "lrshift",
  LRCtrl: "lrctrl",
  PageUpDown: "pageupdown",
  MinusEqual: "minus_equal",
  Brackets: "brackets",
  ShiftTab: "shift_tab",
  Tab: "tab",
  Arrows: "arrows",
} as const;
export type PairGroupValue = (typeof PairGroup)[keyof typeof PairGroup];

// ============================================================
// UI 子枚举（前端独有，未在 Go 端枚举类型化，但仍在多处使用）
// ============================================================

// 状态提示显示模式
export const StatusDisplayMode = {
  Temp: "temp",
  Always: "always",
} as const;
export type StatusDisplayModeValue =
  (typeof StatusDisplayMode)[keyof typeof StatusDisplayMode];

// 状态提示方案名风格
export const SchemaNameStyle = {
  Short: "short",
  Full: "full",
} as const;
export type SchemaNameStyleValue =
  (typeof SchemaNameStyle)[keyof typeof SchemaNameStyle];

// 状态提示位置模式
export const StatusPositionMode = {
  FollowCaret: "follow_caret",
  Custom: "custom",
} as const;
export type StatusPositionModeValue =
  (typeof StatusPositionMode)[keyof typeof StatusPositionMode];

// 数字小键盘行为
export const NumpadBehavior = {
  Direct: "direct",
  FollowMain: "follow_main",
} as const;
export type NumpadBehaviorValue =
  (typeof NumpadBehavior)[keyof typeof NumpadBehavior];

// Shift+字母临时英文行为
export const ShiftBehavior = {
  TempEnglish: "temp_english",
  DirectCommit: "direct_commit",
} as const;
export type ShiftBehaviorValue =
  (typeof ShiftBehavior)[keyof typeof ShiftBehavior];

// ============================================================
// Wails 前端事件名（对应 wind_input/pkg/rpcapi/types.go 中的 WailsEventXxx）
// ============================================================

export const WailsEvent = {
  Config: "config-event",
  Dict: "dict-event",
  Stats: "stats-event",
  System: "system-event",
} as const;
export type WailsEventValue = (typeof WailsEvent)[keyof typeof WailsEvent];
